### PR TITLE
week 2 assessment. I have been sick and ended up not being able to tur…

### PR DESCRIPTION
**Pull request by borja-x13**
…n it in earlier

# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: coding-concepts.js [hunk 1]
```diff
@@ -8,36 +8,42 @@
 // --------------------1) What will this log?
 
 const cohort = "Bravo 2023"
-// console.log(cohort.split(""))
+console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: {"B", "r", "a", "v", "o"," ","2","0","0","3"}
```
Comment (Genemichael25) [inline coding-concepts.js:13]: Right on! Nice job accounting for the space, just make sure we use [] for an array.

File: coding-concepts.js [hunk 2]
```diff
@@ -8,36 +8,42 @@
...
 // --------------------2) What will this log?
 
 const greeter = (name) => {
   `Hello, ${name}!`
 }
+console.log(greeter("LEARN Student"))
+
+// a) Your answer: Hello, LEARN Student
+// b) Verify and explain: Nothing was logged because there is no return statement in the function, which I overlooked. In order to actually get what I expected we need to change the function to the following:
+
```
Comment (Genemichael25) [inline coding-concepts.js:25]: Awesome follow-up! 👏🏽

File: interview-questions.md [hunk 3]
```diff
@@ -8,50 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Your answer: A parameter is part of the structure of a function. The parameters of a function tell the user what values or types of values it can accept and/or are required for the function to work. The arguments are the specific values that the user are giving to the function for processing. for example:
 
-Researched answer:
+```js
+const functionExample = (stringParameter) => {
+    return console.log(`The string parameter (${stringParameter}) is passed as an argument.`)
+}
+
+functionExample("this argument")
+
```
Comment (Genemichael25) [inline interview-questions.md:19]: Love the use of an example here!

File: interview-questions.md [hunk 4]
```diff
@@ -8,50 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: A parameter is a variable of a function and the argument is the data passed in the the given parameters. Arguments make functions more reusable as they can even allow us to pass functions into other functions as arguments
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: value, index and array. Only the value is required, the index and array parameters are optional 
 
-Researched answer:
+Researched answer: element, index, array. The element is what is being processed in the array.map(), the index is the index of the specific element being processed and the array is the array that called .map()
```
Comment (Genemichael25) [inline interview-questions.md:29]: Great job expanding on the initial answer.

File: jest-assessment-week-2-borja/code-challenges.test.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+// it "takes an object { number: 2} and returns if the number is divisible by three"
+// expect(divisibleByThree({number : 15})).toEqual("15 is divisible by three.")
+// expect(divisibleByThree({number : 0})).toEqual("0 is divisible by three.")
+// expect(divisibleByThree({number : -1})).toEqual("-7 is not divisible by three.")
+// input: since there isn't a function to test, there is no input and I expect a fail (red)
+// output: should output a ReferenceError: divisibleByThree is not defined
+ 
+describe ("divisibleByThree", () => {
+    it ("takes an object { number: 2} and returns if the number is divisible by three", () => {
+        expect(divisibleByThree({number : 15})).toEqual("15 is divisible by three.")
```
Comment (Genemichael25) [inline jest-assessment-week-2-borja/code-challenges.test.js:62]: Test is set up nicely, you can use the variables just like you would when invoking a function and passing in the argument.

File: jest-assessment-week-2-borja/code-challenges.test.js [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+//   ● capitalized › takes an array of words and returns an array with the first letter of each word capitalized
+//     ReferenceError: capitalized is not defined
+
+
+// b) Create the function that makes the test pass.
+
+// pseudo: 
+// I will declare a function capitalized that accepts a parameter of an array of words
+// Function will use .map() function to iterate through the array and change each string within to uppercase the first letter
+// Use conditional statements if/else it to return divisible ot not divisible
```
Comment (Genemichael25) [inline jest-assessment-week-2-borja/code-challenges.test.js:131]: How will you target that first letter? You could add that to your pseudocode and I don't think we are using conditional statement for this one.

File: jest-assessment-week-2-borja/code-challenges.test.js [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+// output: index of the first vowel found in the given string
+// yarn jest should Pass (green)
+
+const vowelIndex = (string) => {
+    for (let i in string) {
+        if(string[i] === "a" || string[i] === "e" || string[i] === "i" || string[i] === "o" || string[i] === "u") {
+           return string.indexOf(string[i])
+        }
+    }
+}
```
Comment (Genemichael25) [inline jest-assessment-week-2-borja/code-challenges.test.js:200]: Awesome job! Now we can start to consider what we want to happen if there is no vowel in the string.

File: jest-assessment-week-2-borja/code-challenges.test.js [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+
+//PASS  Desktop/week-2-assessment-borja-x13/jest-assessment-week-2-borja/code-challenges.test.js
+// Tests:       3 passed, 3 total
+
+//------------------------------------ Issues ------------------------------------
+
+// whenever I run yarn jest, I am getting a bunch of unrelated passed and failed results that have nothing to do with my assessment.
```
Comment (Genemichael25) [inline jest-assessment-week-2-borja/code-challenges.test.js:207]: We definitely need to look into this and get it resolved.

**Review by Genemichael25 — APPROVED** [review]
Amazing work on this assessment Borja! Continue to expand on those initial answers in the interview questions section to get you used to verbally demonstrate your knowledge. I left a few comments throughout for you to consider but this is great work. Make sure we address the issue you are having with jest because that is not something we want to continue. If you have any questions about the comments I left, please let me know. This currently displays a competent level of comprehension. Keep it up!

File: jest-assessment-week-2-borja/code-challenges.test.js (hunk 6 above)
Comment (borja-x13) [reply jest-assessment-week-2-borja/code-challenges.test.js:131]: Hey Gene you are right. I started going in one direction in my head during pseudocode and later realized I didn't need any kind of conditional, but forgot to go back and edit it. I will fix that up. Thank you again.

**Pull request merged**